.appveyor.yml

```diff
@@ -88,13 +90,13 @@ install:
       }
   - if "%TOOLCHAIN%"=="mingw" bash -c "echo 'c:/mingw /mingw' >> /etc/fstab"
   - if "%TOOLCHAIN%"=="mingw" bash -c 'mingw-get install libz-dev'
-  - if "%TOOLCHAIN%"=="mingw64" bash -c 'pacman -Sy --noconfirm zlib-devel'
+  - if "%TOOLCHAIN%"=="mingw64" bash -c 'pacman -Sy --noconfirm --needed base-devel libuuid-devel mingw-w64-x86_64-graphviz doxygen mingw-w64-x86_64-doxygen doxygen swig zlib-devel'
```

Comment:
> ... `doxygen mingw-w64-x86_64-doxygen doxygen` ...

Should doxygen really be listed twice like that?

Alnd what's the difference between doxygen and mingw-w64-x86_64-doxygen?  Do we really need to install both?

Installing the `swig` package here seems wrong - we deliberately have bootstrap install a particular version of SWIG (specified by a git commit hash) so that we're using a known-good version of SWIG.  We've found that we sometimes end up needing a more recent SWIG (for a particular bug fix or new feature) than is readily available as a package on common platforms.  E.g. currently git master is using an unreleased snapshot of SWIG because the most recent release doesn't work with Ruby 2.7.  Using the version of SWIG mingw64 have packaged may work OK for Go right now, but this really should work like a developer's tree would or it'll cause problems going forwards (e.g. when someone hooks up Ruby bindings for the appveyor builds).

Reply:
Why are you installing `libuuid-devel`?  We should generate UUIDs via a win32 API call under mingw.

Reply:
I would change the swig part soon,..there are many parts I need to clean up...I will complete them asap.. and for lu

> > ... `doxygen mingw-w64-x86_64-doxygen doxygen` ...
> 
> Should doxygen really be listed twice like that?
> 
> Alnd what's the difference between doxygen and mingw-w64-x86_64-doxygen? Do we really need to install both?
> 
> Installing the `swig` package here seems wrong - we deliberately have bootstrap install a particular version of SWIG (specified by a git commit hash) so that we're using a known-good version of SWIG. We've found that we sometimes end up needing a more recent SWIG (for a particular bug fix or new feature) than is readily available as a package on common platforms. E.g. currently git master is using an unreleased snapshot of SWIG because the most recent release doesn't work with Ruby 2.7. Using the version of SWIG mingw64 have packaged may work OK for Go right now, but this really should work like a developer's tree would or it'll cause problems going forwards (e.g. when someone hooks up Ruby bindings for the appveyor builds).

I would change this soon..there are still many parts for clean up ..I would finish them up soon ..

Reply:
> Why are you installing `libuuid-devel`? We should generate UUIDs via a win32 API call under mingw.

it failed saying this --> https://ci.appveyor.com/project/srinivasyadav18/xapian/builds/32381253
so after installing libuuid-devel it worked

Reply:
It seems we aren't detecting that the platform is "win32" there.  Debugging that...

Reply:
I can reproduce.

The problem is that the `--host` you pass doesn't match the prefix on an install compiler so configure picks unprefixed `gcc` which is for msys2, essentially it's a modified cygwin compiler, so doesn't define `_WIN32`, etc:

```
[00:00:06] $env:CONFIGOPTS+=" --host=x86_64-w64-mingw32"
```
...
```
[00:05:31] checking for x86_64-w64-mingw32-gcc... no
[00:05:31] checking for gcc... gcc
```

And `gcc -v` reports:

```
[00:05:28] COLLECT_GCC=gcc
[00:05:28] COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-msys/11.3.0/lto-wrapper.exe
[00:05:28] Target: x86_64-pc-msys
[00:05:28] Configured with: /c/S/gcc/src/gcc-11.3.0/configure --build=x86_64-pc-msys --prefix=/usr --libexecdir=/usr/lib --enable-bootstrap --enable-shared --enable-shared-libgcc --enable-static --enable-version-specific-runtime-libs --with-arch=x86-64 --with-tune=generic --disable-multilib --enable-__cxa_atexit --with-dwarf2 --enable-languages=c,c++,fortran,lto --enable-graphite --enable-threads=posix --enable-libatomic --enable-libgomp --disable-libitm --enable-libquadmath --enable-libquadmath-support --disable-libssp --disable-win32-registry --disable-symvers --with-gnu-ld --with-gnu-as --disable-isl-version-check --enable-checking=release --without-libiconv-prefix --without-libintl-prefix --with-system-zlib --enable-linker-build-id --with-default-libstdcxx-abi=gcc4-compatible --enable-libstdcxx-filesystem-ts
[00:05:28] Thread model: posix
[00:05:28] Supported LTO compression algorithms: zlib
[00:05:28] gcc version 11.3.0 (GCC) 
```

Checking the preprocessor symbols:

```
[00:06:16] bash -c 'echo __WIN32__ _WIN32 WIN32 __WIN64__ _WIN64 WIN64 __CYGWIN__|gcc -E -'
[00:06:16] # 0 "<stdin>"
[00:06:16] # 0 "<built-in>"
[00:06:16] # 0 "<command-line>"
[00:06:16] # 1 "<stdin>"
[00:06:16] __WIN32__ _WIN32 WIN32 __WIN64__ _WIN64 WIN64 1
```

So it defines `__CYGWIN__` but not the others.

Essentially this CI job is misconfigured because it's not testing what we want to test here (which is mingw64), but it should ideally be clearer what's going on - it took me a while to work it out and I'm more familiar with Xapian's build system than most.  I'm not entirely sure how to improve that as such though.

I also think we want to use `UuidCreate()` in this configuration (we do in real cygwin) - I think that just needs `*msys*` adding as an alternative here:

```
    *-yes | *cygwin*) dnl Use built-in win32 API.
```

I'll do that and also check other places that special case `cygwin` like this.